### PR TITLE
Centralize and simplify checking the screen size

### PR DESCRIPTION
**Pull request by csillag**
This PR doesn't change any functionality, just simplifies how we calculate the `isMobile`, `isTabled` and `isDesktop` flags.

### CODE REVIEW

File: src/app/hooks/useScreensize.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import useMediaQuery from '@mui/material/useMediaQuery'
+import { useTheme } from '@mui/material/styles'
+import { Theme } from '@mui/material/styles/createTheme'
+
+export const useScreenSizeOnTheme = (theme: Theme) => ({
+  isMobile: useMediaQuery(theme.breakpoints.down('sm')),
+  isTablet: useMediaQuery(useTheme().breakpoints.down('md')),
```
Comment (buberdds) [inline src/app/hooks/useScreensize.ts:7]: There is no consistency, `isMobile` relies on function arg and `itTablet`/`isDesktop` use MUI hook directly.

Comment (csillag) [reply src/app/hooks/useScreensize.ts]: Ah sorry, fixed.

File: src/app/hooks/useScreensize.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import useMediaQuery from '@mui/material/useMediaQuery'
+import { useTheme } from '@mui/material/styles'
+import { Theme } from '@mui/material/styles/createTheme'
+
+export const useScreenSizeOnTheme = (theme: Theme) => ({
```
Comment (buberdds) [inline src/app/hooks/useScreensize.ts:5]: Do we need this additional hook? If this was created just to handle `ResultListFrame`. I would just replace `isMobile` in styled component with ` [theme.breakpoints.down('sm')]` syntax (no need for calling hook).

Comment (csillag) [reply src/app/hooks/useScreensize.ts]: OK, done as suggested.

**Review by buberdds — APPROVED** [review]

File: src/app/hooks/useScreensize.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import useMediaQuery from '@mui/material/useMediaQuery'
+import { useTheme } from '@mui/material/styles'
+
+export const useScreenSize = () => {
+  const theme = useTheme()
+  return {
+    isMobile: useMediaQuery(theme.breakpoints.down('sm')),
+    isTablet: useMediaQuery(theme.breakpoints.down('md')),
```
Comment (lukaw3d) [inline src/app/hooks/useScreensize.ts:8]: This seems wrong. Should be between mobile and desktop

Comment (csillag) [reply src/app/hooks/useScreensize.ts:8]: After discussing the same issue with @lubej, I have opened https://github.com/oasisprotocol/explorer/issues/555.

**Pull request merged**